importer/tests/test_views.py

```diff
@@ -20,17 +23,113 @@
 
+    import_url = "/repositories/{0}/import/"
+    repository_url = "/repositories/{0}/"
+
     def test_upload_get(self):
         """GET upload page."""
-        resp = self.client.get("/repositories/{0}/import".format(
+        resp = self.client.get(self.import_url.format(
             self.repo.slug), follow=True)
         body = resp.content.decode("utf-8")
         self.assertTrue('enctype="multipart/form-data"' in body)
 
     def test_upload_get_bad_repo(self):
         """GET upload page."""
-        resp = self.client.get("/repositories/999/import", follow=True)
-        self.assertTrue(resp.status_code == UNAUTHORIZED)
+        self.assert_status_code(self.import_url.format("999"), NOT_FOUND)
+
+    def test_upload_with_permissions(self):
+        """GET upload page with different user permissions"""
+        self.logout()
+        self.login(self.USERNAME_NO_REPO)
+        # user has no permissions at all
+        # user cannot see the repository page
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # and cannot see the import page
+        self.assert_status_code(
+            self.import_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # user has author permissions
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_AUTHOR
+        )
+        # user can see the repository page
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            HTTP_OK
+        )
+        # but cannot see the import for the repo
+        self.assert_status_code(
+            self.import_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # user has no permissions
+        remove_user_from_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_AUTHOR
+        )
+        # User cannot see anything
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        self.assert_status_code(
+            self.import_url.format(self.repo.slug),
+            UNAUTHORIZED
+        )
+        # user has curator permissions
+        assign_user_to_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_CURATOR
+        )
+        # user can see the repository page
+        self.assert_status_code(
+            self.repository_url.format(self.repo.slug),
+            HTTP_OK
+        )
+        # and can see the the import for the repo
+        self.assert_status_code(
+            self.import_url.format(self.repo.slug),
+            HTTP_OK
+        )
+        # remove curator permissions
+        remove_user_from_repo_group(
+            self.user_norepo,
+            self.repo,
+            GroupTypes.REPO_CURATOR
+        )
+        # User cannot see anything
```

Comment:
This check for the user unable to see anything is in this test three times. I think it would be cleaner if it was declared as a function _within_ `test_upload_with_permissions`, and just called three times.